src/errors.rs

```diff
@@ -35,64 +35,55 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-use std::{io, ffi, num};
 use std::fmt::{self, Formatter, Display};
+use std::error::Error;
+use TracerState;
 
 #[derive(Debug)]
 pub enum HWTracerError {
-    // Wrapped errors from elsewhere.
-    FFIIntoString(ffi::IntoStringError),
-    FFINul(ffi::NulError),
-    IO(io::Error),
-    NumParseInt(num::ParseIntError),
-    // Our own errors.
-    CFailure,
-    HardwareSupport(String),
-    InvalidFileName(String),
-    TracerAlreadyStarted,
-    TracerDestroyed,
-    TracerNotStarted,
-    TracingNotPermitted(String),
-}
-
-impl From<ffi::IntoStringError> for HWTracerError {
-    fn from(err: ffi::IntoStringError) -> Self {
-        HWTracerError::FFIIntoString(err)
-    }
-}
+    // Non-fatal errors.
+    HWBufferOverflow,     // The trace buffer being used by the hardware overflowed.
+                          // This is considered a non-fatal error since retrying the tracing may
+                          // succeed.
+    HWSupport(String),    // The hardware doesn't support a required feature. Not fatal for the
```

Comment:
This sounds like it's a positive, but it's a negative, so I think we want something like `NotHWSupported`?

Reply:
How about `NoHWSupport`?

Reply:
Works for me.